string-manipulation-3/src/reverse-words.js

```diff
@@ -0,0 +1,19 @@
+/* exported reverseWords */
+/* result = a string with each character in word reversed. words are in original order.
+create variable to hold new string.
+split input string at each space ' ' and assign to new string.
+each word is now an index in array.
+iterate through each index with for loop
+  for each index (representing one word), split word into characters, reverse index, and join again. concatenate with a space.
+  assign to new reversed string. use addition assignment to add word to new string.
+return reversed string and trim off white space to account for extra spaces.
+*/
+
+function reverseWords(string) {
+  var reverseString = '';
+  var stringArray = string.split(' ');
+  for (var i = 0; i < stringArray.length; i++) {
+    reverseString += stringArray[i].split('').reverse().join('') + ' ';
```

Comment:
By chaining the `split`, `reverse`, and `join` methods here, you are doing _**three full loops**_ over this string. You can do this in one loop if you write the `for` loop yourself

Reply:
![Screen Shot 2022-11-07 at 3 56 33 PM](https://user-images.githubusercontent.com/113560956/200440874-bce1fd0a-f213-4f24-9c70-fb44dfa88f2c.png)